src/nameresolution/builtin.rs

```diff
@@ -70,8 +77,18 @@ pub fn import_prelude<'a>(resolver: &mut NameResolver, cache: &mut ModuleCache<'
 
     // Manually insert some builtins as if they were defined in the prelude
     resolver.current_scope().traits.insert("Int".into(), cache.int_trait);
+
+    // Add string to scope
+    resolver.current_scope().types.insert("string".to_string(), STRING_TYPE);
+    resolver.current_scope().definitions.insert("string".to_string(), BUILTIN_ID);
+
+    // Add pair (,) to scope
     resolver.current_scope().types.insert(Token::Comma.to_string(), PAIR_TYPE);
     resolver.current_scope().definitions.insert(Token::Comma.to_string(), PAIR_ID);
+
+    // Add Ptr to scope
+    resolver.current_scope().types.insert("Ptr".to_string(), PTR_TYPE);
+    resolver.current_scope().definitions.insert("Ptr".to_string(), PTR_ID);
```

Comment:
One last thing - we shouldn't expose the constructor of `Ptr` here since it says we can create a Ptr by packing any type `a` into a struct which isn't true since the runtime representation of that struct isn't behind a pointer. We can probably avoid defining the `TypeConstructor` for the Ptr type completely.

Reply:
It seems to work just without the whole `define_ptr` function with definition pushing. Should i just remove that alltogether now that `ptr` is a built in type like string? Or does it still need to be in `definition_infos`?

Reply:
Only the constructor needs to be in `definition_infos` and since we don't want one for `Ptr` it's fine to remove `define_ptr` altogether